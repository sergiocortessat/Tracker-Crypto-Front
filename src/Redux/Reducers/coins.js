const initialState = {
  coins: [],
};

const coinReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'ALL_COINS':
      return {
        ...state,
        coins: action.payload,
      };
    default:
      return state;
  }
};

export default coinReducer;
