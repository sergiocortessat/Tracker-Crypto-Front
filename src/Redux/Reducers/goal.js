const initialState = {
  goals: [],
};

const goalReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'ALL_GOALS':
      return {
        ...state,
        goal: action.payload,
      };
    default:
      return state;
  }
};

export default goalReducer;
