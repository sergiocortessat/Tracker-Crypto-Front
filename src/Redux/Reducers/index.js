import { combineReducers } from 'redux';
import Coins from './coins';
import User from './user';

const reducers = combineReducers({
  coins: Coins,
  user: User,
});

export default reducers;
