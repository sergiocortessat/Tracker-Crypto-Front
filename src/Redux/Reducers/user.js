const initialState = {
  user: [],
};

const userReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'CURRENT_USER':
      return {
        ...state,
        user: action.payload,
      };
    default:
      return state;
  }
};

export default userReducer;
