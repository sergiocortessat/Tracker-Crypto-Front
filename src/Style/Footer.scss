footer {
  position: fixed;
  bottom: 0;
  height: auto;
  background-color: #67899c;
  width: 100%;

  .footer {
    display: flex;
    flex-direction: row;
    list-style: none;
    color: #fff;
    justify-content: space-around;

    a {
      margin: 10px 5px;
      text-decoration: none;

      li {
        display: flex;
        flex-direction: column;
        align-items: center;

        svg {
          color: lightgrey;
        }

        p {
          font-size: 0.6rem;
          color: #fff;
        }
      }
    }
  }
}
