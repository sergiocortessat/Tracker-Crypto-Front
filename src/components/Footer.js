/* eslint-disable no-unused-vars */
import React from 'react';
import { Link } from 'react-router-dom';
import '../Style/Footer.scss';
import AddBoxIcon from '@material-ui/icons/AddBox';
import TrendingUpIcon from '@material-ui/icons/TrendingUp';
import PieChartIcon from '@material-ui/icons/PieChart';
import MoreHorizIcon from '@material-ui/icons/MoreHoriz';

const Footer = () => {
  const x = 0;
  return (
    <footer className="footer">
      <ul className="footer">
        <Link to="/add-measurement">
          <li className="add-measurement">
            <AddBoxIcon />
            <p>Add measure</p>
          </li>
        </Link>
        <Link to="/">
          <li className="track-item">
            <TrendingUpIcon />
            <p>Track It</p>
          </li>
        </Link>
        <Link to="/">
          <li className="progress">
            <PieChartIcon />
            <p>Your Progress</p>
          </li>
        </Link>
        <Link to="/profile">
          <li className="more">
            <MoreHorizIcon />
            <p>More</p>
          </li>
        </Link>
      </ul>
    </footer>
  );
};

export default Footer;
