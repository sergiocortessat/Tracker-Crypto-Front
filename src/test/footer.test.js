import { Provider } from 'react-redux';
import { createStore } from 'redux';
import renderer from 'react-test-renderer';
import {
  BrowserRouter,
} from 'react-router-dom';
import Reducers from '../Redux/Reducers';
import Footer from '../components/Footer';

const store = createStore(Reducers);

it('renders correctly', () => {
  const addMeasurement = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Footer />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(addMeasurement).toMatchSnapshot();
});
