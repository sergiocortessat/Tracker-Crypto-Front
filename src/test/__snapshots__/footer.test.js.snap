// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<footer
  className="footer"
>
  <ul
    className="footer"
  >
    <a
      href="/add-measurement"
      onClick={[Function]}
    >
      <li
        className="add-measurement"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
          />
        </svg>
        <p>
          Add measure
        </p>
      </li>
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      <li
        className="track-item"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16 6l2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z"
          />
        </svg>
        <p>
          Track It
        </p>
      </li>
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      <li
        className="progress"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 2v20c-5.07-.5-9-4.79-9-10s3.93-9.5 9-10zm2.03 0v8.99H22c-.47-4.74-4.24-8.52-8.97-8.99zm0 11.01V22c4.74-.47 8.5-4.25 8.97-8.99h-8.97z"
          />
        </svg>
        <p>
          Your Progress
        </p>
      </li>
    </a>
    <a
      href="/profile"
      onClick={[Function]}
    >
      <li
        className="more"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
        <p>
          Profile
        </p>
      </li>
    </a>
  </ul>
</footer>
`;
